#! /bin/sh /usr/share/dpatch/dpatch-run
#
## DP: fixes the behaviour of LDFLAGS="-Wl,--as-needed",
## DP: useful to remove unneeded dependencies.

@DPATCH@

--- gtk-sharp/ltmain.sh.old	2008-02-29 06:36:13.000000000 +0100
+++ gtk-sharp/ltmain.sh	2008-02-29 06:36:48.000000000 +0100
@@ -1794,6 +1794,11 @@
 	arg=`$echo "X$arg" | $Xsed -e "s/^ //"`
 	;;
 
+      -Wl,--as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+
       -Wl,*)
 	args=`$echo "X$arg" | $Xsed -e "$sed_quote_subst" -e 's/^-Wl,//'`
 	arg=
@@ -2134,6 +2139,15 @@
 	lib=
 	found=no
 	case $deplib in
+	-Wl,--as-needed)
+	  if test "$linkmode,$pass" = "prog,link"; then
+	    compile_deplibs="$deplib $compile_deplibs"
+	    finalize_deplibs="$deplib $finalize_deplibs"
+	  else
+	    deplibs="$deplib $deplibs"
+	  fi
+	  continue
+	  ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"
